addons/payment/models/payment_acquirer.py

```diff
@@ -314,6 +314,12 @@ def _get_compatible_acquirers(
             domain = expression.AND([domain, [('allow_tokenization', '=', True)]])
 
         compatible_acquirers = self.env['payment.acquirer'].search(domain)
+
+        if self.env.user._is_public():
+            compatible_acquirers = compatible_acquirers.filtered(
+                lambda acq: not self._is_tokenization_required(provider=acq.provider)
```

Comment:
```suggestion
                lambda acq: not self._is_tokenization_required(provider=acq.provider, **kwargs)
```
Should not be necessary but better to be safe than sorry.